lib/connection/connection_config.js

```diff
@@ -485,12 +485,12 @@ function ConnectionConfig(options, validateCredentials, qaMode, clientInfo)
     disableQueryContextCache = options.disableQueryContextCache;
   }
 
-  let loginTimeout = 300;
-  if (Util.exists(options.loginTimeout)) {
-    Errors.checkArgumentValid(Util.isNumber(options.loginTimeout),
-    ErrorCodes.ERR_CONN_CREATE_INVALID_MAX_LOGIN_TIMEOUT);
+  let retryTimeout = 300;
+  if (Util.exists(options.retryTimeout)) {
+    Errors.checkArgumentValid(Util.isNumber(options.retryTimeout),
+    ErrorCodes.ERR_CONN_CREATE_INVALID_MAX_RETRY_TIMEOUT);
 
-    loginTimeout = Math.max(loginTimeout,options.loginTimeout);
+    retryTimeout = Math.max(retryTimeout,options.retryTimeout);
```

Comment:
Please allow it to be set to 0 as unlimited.

Reply:
fixed